### PR TITLE
change all tag helper descriptors to not use constructors - part 1

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#441 
@NTaylorMullen

### CODE REVIEW

File: test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs [hunk 1]
```diff
@@ -138,10 +138,12 @@ public static TheoryData OutputElementHintData
                                 attributes: Enumerable.Empty<TagHelperAttributeDescriptor>(),
                                 requiredAttributes: Enumerable.Empty<string>(),
                                 tagStructure: default(TagStructure),
-                                designTimeDescriptor: new TagHelperDesignTimeDescriptor(
-                                    summary: null,
-                                    remarks: null,
-                                    outputElementHint: "strong"))
+                                designTimeDescriptor: new TagHelperDesignTimeDescriptor
+                                {
+                                    Summary = null,
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs]: No need to set these values to `null`. Just don't set them. Same applies for other instances.

Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs]: I just thought that explicitly setting it is better in a test.

Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs]: Nah. Just noise :smile:

**Comment by NTaylorMullen** [comment]
:shipit: once https://github.com/aspnet/Razor/pull/488#discussion_r37103379 is completed.

File: test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs (hunk 1 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs]: Okay. Will change

**Comment by dougbu** [comment]
`Equals()` and `GetHashCode()` implementations in these descriptors (if any) plus `TagHelperAttributeDescriptorComparer` and `TagHelperDesignTimeDescriptorComparer` should change now that all properties of these descriptors are settable. Shouldn't really include mutable values in any of that.

**Comment by ajaybhargavb** [comment]
Spoke offline; Decided not to change the comparers.

**Pull request merged**